### PR TITLE
feat: update for Ruff and related updates

### PR DESCRIPTION
**Pull request by henryiii**
Adding Ruff and similar updates.

### CODE REVIEW

**Comment by henryiii** [comment]
@klieret This is ready for review (just in case you didn't get notified after tests were passing, they weren't originally).

File: slides.md [hunk 1]
```diff
@@ -141,19 +137,18 @@ layout: two-cols
 <v-click>
 
 * **Examples**:
-  * [scikit-hep project template](https://github.com/scikit-hep/cookie/): All the features, all the best-practices
-  * [my personal python template](https://github.com/scikit-hep/cookie/): Fewer options, easier to read (I think ;))
+  * [Scientific Python Cookiee](https://github.com/scientific-python/cookie): All the features, all the best-practices
```
Comment (klieret) [inline slides.md:140]: Typo `Cookiee``

**Review by klieret — APPROVED** [review]
Thanks a lot for all of these. You even made a lot of corrections for my part. Highly appreciated!

File: slides.md (hunk 1 above)
Comment (henryiii) [inline slides.md:140]:
```suggestion
  * [Scientific Python Cookie](https://github.com/scientific-python/cookie): All the features, all the best-practices
```

**Pull request merged**